4.7 system cap updates (#128)
smartdevicelink/sdl_android_guides: docs/System Capability Manager/index.md

## Querying Capabilities

> Any point after receiving the first `ON_HMI_STATUS` notification from the connected module, you can access the `SystemCapability` manager and its data. Your instance of `SdlManager` will have convenience methods that tie into the `SystemCapabilityManager`.

Reviewer: ```suggestion
Any point after receiving the first `ON_HMI_STATUS` notification from the connected module, you can access the `SystemCapability` manager and its data. Your instance of `SdlManager` will provide access to the `SystemCapabilityManager`.
```

```java
sdlManager.getSystemCapabilityManager().getCapability(SystemCapabilityType.BUTTON, new OnSystemCapabilityListener(){

    @Override
    public void onCapabilityRetrieved(Object capability){
        ButtonCapabilities buttonCapabilities = (ButtonCapabilities) capability;

>         SystemCapabilityManager.convertToList(capability, ButtonCapabilities.class);

Reviewer: This new code is wrong. I casts a list into a single object then just calls the `convertToList` method but the return value isn't saved. The old code worked, why was this changed?

    }

    @Override
    public void onError(String info){
        Log.i(TAG, "Capability could not be retrieved: "+ info);
    }
});
```

This method prevents the developer from having to suppress a warning as well as creates a safe way to cast the object to a list.

## Asynchronous vs Synchronous Queries

> Some capabilities will be instantly available after the first `ON_HMI_STATUS` notification. These are parsed from the `RegisterAppInterface` response. However, some capabilities MUST be obtained asynchronously and therefore require a callback to be obtained.  If a capability can be retrieved synchronously another method can be used via the `SdlManager` object, `sdlManager.getSystemCapabilityManager().getCapability(SystemCapabilityType)`.

Reviewer: ```suggestion
Some capabilities will be instantly available after the first `OnHMIStatus` notification. These are parsed from the `RegisterAppInterface` response. However, some capabilities MUST be obtained asynchronously and therefore require a callback to be obtained.  If a capability can be retrieved synchronously another method can be used via the `SystemCapbilityManager` object obtained from the `SdlManager`, `sdlManager.getSystemCapabilityManager().getCapability(SystemCapabilityType)`.
```
